app/scripts/scripts.js

```diff
@@ -33,16 +13,94 @@ var hangmanWords = [
    "come","made","may","part"
 ];
 
+//creates an array of easy words (difficulty based on word length)
 var easyArray = hangmanWords.filter(function(word){
   return word.length <= 4;
 });
 
+//creates an array of hard words (difficulty based on word length)
 var hardArray = hangmanWords.filter(function(word){
   return word.length > 4;
 });
+//sets the word to be guessed
+var currentWordFull = easyArray[47],//IMPORTANT: replace the number with wordSelect (the function) for production use
+//set an all lower case version of the current word
+currentWord = currentWordFull.toLowerCase();
 
+//creates blocks in the DOM indicating where there are letters and spaces
+currentWord.split("").map(function(elem) {
+  var guessWordBlock = document.getElementById("word-to-guess");
+
+  var domElem = document.createElement("div");
+
+  if(elem.match(/[a-z]/i)) {
+    domElem.className = "character-block is-letter";
+  } else {
+    domElem.className = "character-block";
+  }
+
+  //console.log(domElem);//for testing, remove later
+  guessWordBlock.appendChild(domElem);
+});
+
+/* For picking letters from alphabet keypad. Returns the
+   letter chosen or "already picked"
+*/
+function pickLetter(letter) {
+  "use strict";
+  var result;
+  //checks if the class name is still a button or not
+  if (letter.className === "letter-button") {
+    //if it's still a button it returns the letter
+    //then disables the button
+    result = letter.innerHTML;
+    letter.className = "letter-disabled";
+
+    //calls this function to deal with where the letter goes
+    handlePickedLetter(result);
+  } else {
+    //if the buttons is disabled it logs this
+    result = "already picked";
+  }
+  // for testing, remove later
+  console.log(result);
+}
+
+//picks out a word in the "hangmanWords" array and returns it
 function wordSelect (array) {
   var num = Math.floor(Math.random() * array.length);
```

Comment:
This should be array.length - 1   arrays are zero based index, so at the high end, the number may be outside the array range.